packages/shared_preferences/shared_preferences_web/test/shared_preferences_web_test.dart

```diff
@@ -26,11 +25,8 @@ void main() {
     });
 
     test('registers itself', () {
-      expect(SharedPreferencesStorePlatform.instance,
```

Comment:
I removed this test because it was making the test fail and by reading `SharedPreferences.registerWith(registrar)`'s implementation, it always sets the instance to `SharedPreferencesPlugin`.
https://github.com/flutter/plugins/blob/2e22820df8f7b9f4cd5bd7eb2c3ccd19342501ca/packages/shared_preferences/shared_preferences_web/lib/shared_preferences_web.dart#L17-L19

Maybe I am missing something.

Reply:
@ditman ^^^

Reply:
Looking at this again: yes, it always registers itself as a `SharedPreferencesPlugin`, but this check is *before* the registration, making sure that the later check is actually testing something meaningful.

What exactly was the error? Does this just need handling of a null instnace?

Reply:
True, as per here : https://github.com/flutter/plugins/blob/2e22820df8f7b9f4cd5bd7eb2c3ccd19342501ca/packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart#L38-L39

It is supposed to return a `MethodChannelSharedPreferencesStore` but the test fails for some reason :

```
00:00 +0: SharedPreferencesPlugin registers itself
00:00 +0: SharedPreferencesPlugin registers itself [E]
  Expected: not <Instance of 'SharedPreferencesPlugin'>
    Actual: <Instance of 'SharedPreferencesPlugin'>

  dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49  throw_
  packages/test_api/src/frontend/expect.dart 155:31                             fail
  packages/test_api/src/frontend/expect.dart 150:3                              _expect
  packages/test_api/src/frontend/expect.dart 59:3                               expect$
  packages/flutter_test/src/widget_tester.dart 429:3                            expect$
  shared_preferences_web_test.dart 29:7                                         <fn>
  packages/test_api/src/backend/declarer.dart 200:19                            <fn>
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>
  packages/stack_trace/src/stack_zone_specification.dart 126:26                 <fn>
  packages/stack_trace/src/stack_zone_specification.dart 208:15                 [_run]
  packages/stack_trace/src/stack_zone_specification.dart 126:14                 <fn>
  dart-sdk/lib/async/zone.dart 1362:47                                          _rootRunUnary
  dart-sdk/lib/async/zone.dart 1265:19                                          runUnary
  dart-sdk/lib/async/future_impl.dart 152:18                                    handleValue
  dart-sdk/lib/async/future_impl.dart 704:44                                    handleValueCallback
  dart-sdk/lib/async/future_impl.dart 733:13                                    _propagateToListeners
  dart-sdk/lib/async/future_impl.dart 403:9                                     <fn>
  packages/stack_trace/src/stack_zone_specification.dart 208:15                 [_run]
  packages/stack_trace/src/stack_zone_specification.dart 116:48                 <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1162:7                                           runGuarded
  dart-sdk/lib/async/zone.dart 1202:23                                          callback
  dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
  dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15           <fn>
  ===== asynchronous gap ===========================
  dart-sdk/lib/async/zone.dart 1286:19                                          registerUnaryCallback
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 67:19            _async
  packages/test_api/src/backend/declarer.dart 198:22                            <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1787:67                                          _runZoned
  dart-sdk/lib/async/zone.dart 1710:10                                          runZoned
  packages/test_api/src/backend/declarer.dart 198:13                            <fn>
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54            runBody
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5            _async
  packages/test_api/src/backend/declarer.dart 181:48                            <fn>
  packages/test_api/src/backend/invoker.dart 231:15                             <fn>
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 84:54            runBody
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 123:5            _async
  packages/test_api/src/backend/invoker.dart 228:14                             <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1787:67                                          _runZoned
  dart-sdk/lib/async/zone.dart 1710:10                                          runZoned
  packages/test_api/src/backend/invoker.dart 228:5                              waitForOutstandingCallbacks
  packages/test_api/src/backend/invoker.dart 383:17                             <fn>
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50            <fn>
  packages/stack_trace/src/stack_zone_specification.dart 126:26                 <fn>
  packages/stack_trace/src/stack_zone_specification.dart 208:15                 [_run]
  packages/stack_trace/src/stack_zone_specification.dart 126:14                 <fn>
  dart-sdk/lib/async/zone.dart 1362:47                                          _rootRunUnary
  dart-sdk/lib/async/zone.dart 1265:19                                          runUnary
  dart-sdk/lib/async/future_impl.dart 152:18                                    handleValue
  dart-sdk/lib/async/future_impl.dart 704:44                                    handleValueCallback
  dart-sdk/lib/async/future_impl.dart 733:13                                    _propagateToListeners
  dart-sdk/lib/async/future_impl.dart 530:7                                     [_complete]
  dart-sdk/lib/async/future.dart 174:15                                         <fn>
  packages/stack_trace/src/stack_zone_specification.dart 208:15                 [_run]
  packages/stack_trace/src/stack_zone_specification.dart 116:48                 <fn>
  dart-sdk/lib/async/zone.dart 1346:47                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1162:7                                           runGuarded
  dart-sdk/lib/async/zone.dart 1202:23                                          <fn>
  packages/stack_trace/src/stack_zone_specification.dart 208:15                 [_run]
  packages/stack_trace/src/stack_zone_specification.dart 116:48                 <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1186:23                                          <fn>
  dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart 48:19       internalCallback
  ===== asynchronous gap ===========================
  dart-sdk/lib/async/zone.dart 1286:19                                          registerUnaryCallback
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 67:19            _async
  packages/test_api/src/backend/invoker.dart 370:18                             <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1787:67                                          _runZoned
  dart-sdk/lib/async/zone.dart 1710:10                                          runZoned
  packages/test_api/src/backend/invoker.dart 370:9                              <fn>
  packages/test_api/src/backend/invoker.dart 415:15                             [_guardIfGuarded]
  packages/test_api/src/backend/invoker.dart 369:7                              <fn>
  packages/stack_trace/src/chain.dart 94:24                                     <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1787:67                                          _runZoned
  dart-sdk/lib/async/zone.dart 1710:10                                          runZoned
  packages/stack_trace/src/chain.dart 92:12                                     capture
  packages/test_api/src/backend/invoker.dart 368:11                             [_onRun]
  packages/test_api/src/backend/live_test_controller.dart 153:11                run
  dart-sdk/lib/async/future.dart 198:37                                         <fn>
  dart-sdk/lib/async/zone.dart 1346:47                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1162:7                                           runGuarded
  dart-sdk/lib/async/zone.dart 1202:23                                          <fn>
  dart-sdk/lib/async/zone.dart 1354:13                                          _rootRun
  dart-sdk/lib/async/zone.dart 1258:19                                          run
  dart-sdk/lib/async/zone.dart 1162:7                                           runGuarded
  dart-sdk/lib/async/zone.dart 1202:23                                          callback
  dart-sdk/lib/async/schedule_microtask.dart 40:11                              _microtaskLoop
  dart-sdk/lib/async/schedule_microtask.dart 49:5                               _startMicrotaskLoop
  dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 166:15           <fn>
```

Reply:
True, as per here : https://github.com/flutter/plugins/blob/2e22820df8f7b9f4cd5bd7eb2c3ccd19342501ca/packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart#L38-L39

It is supposed to return a `MethodChannelSharedPreferencesStore` but the test fails on `master` for some reason :

```
00:00 +0: SharedPreferencesPlugin registers itself
00:00 +0: SharedPreferencesPlugin registers itself [E]
  Expected: not <Instance of 'SharedPreferencesPlugin'>
    Actual: <Instance of 'SharedPreferencesPlugin'>

  dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49  throw_
  packages/test_api/src/frontend/expect.dart 155:31                             fail
  packages/test_api/src/frontend/expect.dart 150:3                              _expect
```

Reply:
Odd. How about instead of just removing that line, replacing it with manually setting the `instance` to a dummy value (i.e, an empty subclass created by the test) then. Otherwise in the case you are seeing, nothing is actually being tested.

Reply:
I think this is testing that the instance is null until `SharedPreferencesPlugin.registerWith` is called a couple of lines below.

(Maybe with null safety that part of the test is not required anymore?) We'll eventually have to revisit these tests so they run with flutter drive instead of flutter run --platform chrome. Thanks for the migration!!